tests/run-tests.py

```diff
@@ -0,0 +1,130 @@
+import contextlib
+import logging
+import os
+import subprocess as sp
+import sys
+import time
+from enum import Enum
+
+EXPECTED_TIME_TO_BOOT_WO_ACCEL = 60  # seconds
+RESET = "\033[0m"
+BOLD = "\033[1m"
+RED = "\033[31m"
+OSBUILD = os.environ["OSBUILD"]
+BASE_INPUT = os.environ["BASE_INPUT"]
+BASE_OUTPUT = os.environ["BASE_OUTPUT"]
+WEB_OUTPUT = os.environ["WEB_OUTPUT"]
+QCOW2_OUTPUT = os.environ["QCOW2_OUTPUT"]
+IMAGE_PATH = os.environ["IMAGE_PATH"]
+
+
+logging.basicConfig(level=logging.getLevelName(os.environ.get("TESTS_LOGLEVEL", "INFO")))
+
+
+def run_osbuild(input: str, output: str, pipeline: str, check=True):
+    cmd = [OSBUILD, f"--input={input}", f"--output={output}", pipeline]
+    logging.info(f"Running osbuild: {cmd}")
+    osbuild = sp.run(cmd, capture_output=True)
+    if osbuild.returncode != 0:
+        logging.error("osbuild failed!")
+        print("STDERR")
+        print(osbuild.stderr)
+        print("STDOUT")
+        print(osbuild.stdout)
+        if check:
+            sys.exit(1)
+
+    return osbuild.returncode
+
+
+def build_web_server_image():
+    run_osbuild(BASE_INPUT, BASE_OUTPUT, "1-create-base.json")
+    r = run_osbuild(BASE_OUTPUT, WEB_OUTPUT, "2-configure-web-server.json", check=False)
+    # TODO: remove this workaround once the grub2 stage works on the first try :-)
+    if r != 0:
+        run_osbuild(BASE_OUTPUT, WEB_OUTPUT, "2-configure-web-server.json")
+    run_osbuild(WEB_OUTPUT, QCOW2_OUTPUT, "3-compose-qcow2.json")
+
+
+class QemuAcceleration(Enum):
+    KVM = 0
+    HVF = 1
+    NONE = 2
+
+
+class TestResult(Enum):
+    SUCCESS = 0
+    FAIL = 1
+
+
+def uname() -> str:
+    uname = sp.run(["uname"], capture_output=True)
+    return uname.stdout.decode("utf-8").strip()
+
+def detect_qemu_acceleration() -> QemuAcceleration:
+    system = uname()
+    if system == "Linux":
+        lsmod = p = sp.run(["lsmod | grep kvm"], capture_output=True, shell=True)
+        if lsmod.returncode == 0:
+            logging.info("Detected Linux system with KVM support")
+            return QemuAcceleration.KVM
+        else:
+            logging.info("Detected Linux system w/o KVM support")
+            return QemuAcceleration.NONE
+    elif system == "Darwin":
+        logging.info("Detected macOS (support for hvf is expected)")
+        return QemuAcceleration.HVF
+    else:
+        logging.info("Unknown OS")
+        return QemuAcceleration.NONE
+
+
+@contextlib.contextmanager
+def boot_image(path: str):
+    qemu_accel_options = {
+        QemuAcceleration.KVM: ["-accel", "kvm"],
+        QemuAcceleration.HVF: ["-accel", "hvf"],
+        QemuAcceleration.NONE: []
+    }
+    acceleration = detect_qemu_acceleration()
```

Comment:
Detecting is error-prone and I don't think it's necessary. You can pass `-machine accel=kvm:tcg` to `qemu`, which will try to use `kvm` and if it fails to initialize, it falls back to `tcg` (the software one). I don't think we need to keep it just for timeout. Just make it big enough :)

Reply:
done